scapy/fields.py

```diff
@@ -554,6 +554,38 @@ def getfield(self, pkt, s):
         l = self.length_from(pkt)
         return s[l:], self.m2i(pkt,s[:l])
     
+class XStrField(StrField):
+    """
+    StrField which value is printed as hexadecimal.
+    """
+
+    def i2repr(self, pkt, x):
+        return "".join(c.encode("hex") for c in x)
+
+class XStrLenField(StrLenField):
+    """
+    StrLenField which value is printed as hexadecimal.
+    """
+
+    def i2repr(self, pkt, x):
+        output = ""
+        max_idx = self.length_from(pkt)
+        for i in range(0, max_idx):
```

Comment:
Please do **not** create unneeded lists.
Also, try to keep the code as simple as possible. I think that this would do the job:
```Python
    def i2repr(self, pkt, x):
        return x[:self.length_from(pkt)].encode('hex')
```